File: files/spice/ltspice_data_analysis.m
%% Init add-on
addpath("ltspice2matlab/")

%% Read raw
raw = LTspice2Matlab('./Universal_Biquad_tl082.raw');
t = raw.time_vect;
vin = raw.variable_mat(8,:);
vbp = raw.variable_mat(5,:);
vbs = raw.variable_mat(6,:);


%% Plot
fig1 = figure(1);
plot(t, vin, t, vbp)
grid
xlabel('Time t in s')
ylabel('Voltages in V')
legend('V_in(t)','V_bp(t)')

File: files/spice/ltspice_data_analysis.py
#!/usr/bin/env python
# coding: utf-8
"""Post-processing of ltspice data"""

import ltspice
import matplotlib.pyplot as plt

# %% Load RAW LTSpice into python workspace
file = "./Universal_Biquad_tl082.raw"
raw = ltspice.Ltspice(file)
raw.parse()

# %% Extract voltages and time
t = raw.get_time()
vin = raw.get_data('V(vin)')
vbp = raw.get_data('V(vbp)')

# %% Plots
plt.plot(t, vin, label=r'$v_{in}(t)$')
plt.plot(t, vbp, label=r'$v_{bp}(t)$')
plt.legend()
plt.xlabel('Time t in s')
plt.ylabel('Voltages in V')
plt.grid()
plt.show()
